docs/en/Architecture-DataFlow.md

```diff
@@ -0,0 +1,122 @@
+
+![Data Flow of Read from a Local Worker]({{ site.baseurl }}/img/dataflow-local-cache-hit.gif)
+
+#### Read Hits Alluxio Cache on a Remote Worker
+
+When the data is not available in a local Alluxio worker but resides in another Alluxio worker for the cluster, the Alluxio client will read from the remote worker. For example, if the Alluxio client checks with the Alluxio master and finds that the data is available on a remote Alluxio worker, the local worker will read the data from the remote worker and forward the data to the client.
+
+In addition to returning the data to the client, the worker will also write a copy locally so that future reads of the same data can be served locally from memory. Remote cache hits provide network-speed data reads. Alluxio prioritizes reading from remote workers over reading from under storage because the network speed between Alluxio workers is typically faster than the speed between Alluxio workers and the under storage.
+
+![Data Flow of Read from a Remote Worker]({{ site.baseurl }}/img/dataflow-remote-cache-hit.gif)
+
+
+#### Read Misses Alluxio Cache and Triggers Cache
+
+If the data is not available within the Alluxio space, a cache miss occurs and the application will have to read the data from the under storage. The Alluxio client will delegate the read to a local worker and the worker will read the data from the under storage. Up to the client, the worker may cache the data locally for future reads. Cache misses generally cause the largest delay because the application has to wait until the data is fetched from the under storage. A cache miss typically happens when the data is read the first time.
+
+when Alluxio client reads a block that is not locally available, it will read and cache the block entirely on the local worker even if only part of the block is requested by the client. If there is no such a local worker, no cache will happen. For certain workloads and file formats (e.g., running SQL queries on ORC and Parquet files), the data reads are often not sequential. In this case, having partial caching will help users cache the input data into Alluxio even if only the footer or header of these files are read, significantly improving the performance of the following queries on the same input data.
+
+In Alluxio versions 1.7 and above, partial caching is always enabled and handled on the server side in the background. The caching process is completely transparent to the client. The client will read data as normal and signal to the worker which blocks it reads asynchronously. The worker will fetch these blocks from the under storage or other Alluxio workers asynchronously. Duplicate requests caused by concurrent readers will be consolidated on the worker and result in caching the block once. Partial caching is not on the critical path, but may still impact performance if the network bandwidth between Alluxio and the under storage system is a bottleneck.
```

Comment:
Our docs are versioned and 1.7 came out almost a year ago. Should we simplify this section to only talk about the 1.7/1.8 behavior?